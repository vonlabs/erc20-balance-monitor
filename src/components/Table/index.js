import styled from '@emotion/styled';

// mui
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import Paper from '@mui/material/Paper';
import IconButton from '@mui/material/IconButton';

// mui icons
import DeleteForeverIcon from '@mui/icons-material/DeleteForever';

// custom
import { CryptoIcon } from '../CryptoCoin';

const STableContainer = styled(TableContainer)`
  .MuiTableCell-root {
 //   height: 20px;
  }
  .bold {
    font-weight: 700;
  }
  .coin {
    width: 68px;
  }
  .coin-cell-content{
    display: inline-flex;
  }
  .address {
    width: 468px;
  }
  .remove-btn-cell {
    padding: 4px 16px;
  }
`

const SIconButton =  styled(IconButton)`
  padding: 8px;
`

export default function CoinTable(props) {
  return (
    <STableContainer component={Paper}>
      <Table aria-label="coin table">
        <TableHead>
          <TableRow>
            <TableCell className='bold coin'>Coin</TableCell>
            <TableCell className='bold address'>Address</TableCell>
            <TableCell className='bold'>Balance</TableCell>
            <TableCell></TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {props.addresses.map((row) => (
            <TableRow
              key={row.uuid}
              sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
            >
              <TableCell component="th" scope="row">
                <div className="coin-cell-content">
                  <CryptoIcon className={row.coin}/> {row.coin}
                </div>
              </TableCell>
              <TableCell>{row.address}</TableCell>
              <TableCell>
                {props.balances[row.uuid]}
              </TableCell>
              <TableCell 
                align="right"
                className="remove-btn-cell"
              >
                <SIconButton
                  aria-label="remove address" 
                  component="span"
                  onClick={()=>{props.removeAddress(row.uuid)}}
                >
                  <DeleteForeverIcon />
                </SIconButton>
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </STableContainer>
  );
}