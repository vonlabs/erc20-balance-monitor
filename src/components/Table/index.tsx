import styled from "@emotion/styled";

// mui
import Table from "@mui/material/Table";
import TableBody from "@mui/material/TableBody";
import TableCell from "@mui/material/TableCell";
import TableContainer from "@mui/material/TableContainer";
import TableHead from "@mui/material/TableHead";
import TableRow from "@mui/material/TableRow";
import IconButton from "@mui/material/IconButton";

// mui icons
import DeleteForeverIcon from "@mui/icons-material/DeleteForever";

// custom
import { CryptoIcon } from "../CryptoCoin";

// types
import { Uuid, AddressObj, BalanceObj } from "../../types";

const STableContainer = styled(TableContainer)`
  .bold {
    font-weight: 700;
  }
  .coin {
    width: 68px;
  }
  .coin-cell-content {
    display: inline-flex;
  }
  .address {
    width: 468px;
  }
  .remove-btn-cell {
    padding: 4px 16px;
  }
`;

const SIconButton = styled(IconButton)`
  padding: 8px;
`;

type Props = {
  removeAddress: (uuid: Uuid) => void;
  addresses: AddressObj[];
  balances: BalanceObj;
};

export default function CoinTable(props: Props): JSX.Element {
  return (
    <STableContainer>
      <Table aria-label="coin table">
        <TableHead>
          <TableRow>
            <TableCell className="bold coin">Coin</TableCell>
            <TableCell className="bold address">Address</TableCell>
            <TableCell className="bold">Balance</TableCell>
            <TableCell></TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {props.addresses.map((row) => (
            <TableRow
              key={row.uuid}
              sx={{ "&:last-child td, &:last-child th": { border: 0 } }}
            >
              <TableCell component="th" scope="row">
                <div className="coin-cell-content">
                  <CryptoIcon className={row.coin} /> {row.coin}
                </div>
              </TableCell>
              <TableCell>{row.address}</TableCell>
              <TableCell>{props.balances[row.uuid]}</TableCell>
              <TableCell align="right" className="remove-btn-cell">
                <SIconButton
                  aria-label="remove address"
                  onClick={() => {
                    props.removeAddress(row.uuid);
                  }}
                >
                  <DeleteForeverIcon />
                </SIconButton>
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </STableContainer>
  );
}
