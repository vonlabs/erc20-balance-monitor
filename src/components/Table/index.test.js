import React from 'react';
import renderer from 'react-test-renderer';
import Table from "./index";

const addres = [{
  "uuid": "d49d2cbb-9320-42ce-b234-83b6f15579c2",
  "coin": "LINK",
  "address": "0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1"
}]

const addreses = [{
  "uuid": "d49d2cbb-9320-42ce-b234-83b6f15579c1",
  "coin": "LINK",
  "address": "0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1"
},{
  "uuid": "d49d2cbb-9320-42ce-b234-83b6f15579c2",
  "coin": "ETH",
  "address": "0x90f8bf6a479f320ead074411a4b0e7944ea8c9c2"
},{
  "uuid": "d49d2cbb-9320-42ce-b234-83b6f15579c3",
  "coin": "LINK",
  "address": "0x90f8bf6a479f320ead074411a4b0e7944ea8c9c3"
}]

const balances = {
  "d49d2cbb-9320-42ce-b234-83b6f15579c1": "0123",
  "d49d2cbb-9320-42ce-b234-83b6f15579c2": "4567",
  "d49d2cbb-9320-42ce-b234-83b6f15579c3": "8901"
}

it('Table renders correctly empty table', () => {
  const tree = renderer
    .create(<Table
      addresses={[]}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Table renders correctly table with 1 item', () => {
  const tree = renderer
    .create(<Table
      addresses={addres}
      balances={balances}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Table renders correctly table with multiple item', () => {
  const tree = renderer
    .create(<Table
      addresses={addreses}
      balances={balances}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
