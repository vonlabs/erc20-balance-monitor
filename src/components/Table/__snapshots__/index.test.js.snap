// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly empty table 1`] = `
<div
  className="MuiTableContainer-root css-b3ber7-MuiTableContainer-root"
>
  <table
    aria-label="coin table"
    className="MuiTable-root css-rqglhn-MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium bold coin css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Coin
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium bold address css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Address
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium bold css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Balance
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        />
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      role={null}
    />
  </table>
</div>
`;

exports[`Table renders correctly table with 1 item 1`] = `
<div
  className="MuiTableContainer-root css-b3ber7-MuiTableContainer-root"
>
  <table
    aria-label="coin table"
    className="MuiTable-root css-rqglhn-MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium bold coin css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Coin
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium bold address css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Address
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium bold css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Balance
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        />
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root css-34nofg-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          scope="row"
        >
          <div
            className="coin-cell-content"
          >
            <div
              className="LINK css-9z3aw5"
            />
             
            LINK
          </div>
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium balance-cell css-1ex1afd-MuiTableCell-root"
        >
          <span
            className="mobile-label"
          >
            Balance: 
          </span>
          4567
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium remove-btn-cell css-177gid-MuiTableCell-root"
        >
          <button
            aria-label="remove address"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1kfzivx-MuiButtonBase-root-MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="DeleteForeverIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12 1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Table renders correctly table with multiple item 1`] = `
<div
  className="MuiTableContainer-root css-b3ber7-MuiTableContainer-root"
>
  <table
    aria-label="coin table"
    className="MuiTable-root css-rqglhn-MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium bold coin css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Coin
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium bold address css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Address
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium bold css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Balance
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        />
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root css-34nofg-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          scope="row"
        >
          <div
            className="coin-cell-content"
          >
            <div
              className="LINK css-9z3aw5"
            />
             
            LINK
          </div>
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium balance-cell css-1ex1afd-MuiTableCell-root"
        >
          <span
            className="mobile-label"
          >
            Balance: 
          </span>
          0123
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium remove-btn-cell css-177gid-MuiTableCell-root"
        >
          <button
            aria-label="remove address"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1kfzivx-MuiButtonBase-root-MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="DeleteForeverIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12 1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root css-34nofg-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          scope="row"
        >
          <div
            className="coin-cell-content"
          >
            <div
              className="ETH css-9z3aw5"
            />
             
            ETH
          </div>
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          0x90f8bf6a479f320ead074411a4b0e7944ea8c9c2
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium balance-cell css-1ex1afd-MuiTableCell-root"
        >
          <span
            className="mobile-label"
          >
            Balance: 
          </span>
          4567
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium remove-btn-cell css-177gid-MuiTableCell-root"
        >
          <button
            aria-label="remove address"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1kfzivx-MuiButtonBase-root-MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="DeleteForeverIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12 1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root css-34nofg-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          scope="row"
        >
          <div
            className="coin-cell-content"
          >
            <div
              className="LINK css-9z3aw5"
            />
             
            LINK
          </div>
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          0x90f8bf6a479f320ead074411a4b0e7944ea8c9c3
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium balance-cell css-1ex1afd-MuiTableCell-root"
        >
          <span
            className="mobile-label"
          >
            Balance: 
          </span>
          8901
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium remove-btn-cell css-177gid-MuiTableCell-root"
        >
          <button
            aria-label="remove address"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1kfzivx-MuiButtonBase-root-MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="DeleteForeverIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12 1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
